### PR TITLE
adding model definition for AccountCollection and relations with Account and Collection

### PR DESCRIPTION
**Pull request by jasperboyd**
This change is [Reviewable](https://reviewable.io/reviews/ten24/slatwall/4962)

### CODE REVIEW

File: model/entity/AccountCollection.cfc [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	property name="remoteID" ormtype="string";
+
+	// Audit Properties
+	property name="createdDateTime" hb_populateEnabled="false" ormtype="timestamp";
+	property name="createdByAccountID" hb_populateEnabled="false" ormtype="string";
+	property name="modifiedDateTime" hb_populateEnabled="false" ormtype="timestamp";
+	property name="modifiedByAccountID" hb_populateEnabled="false" ormtype="string";
+
+	// Non-Persistent Properties
+	property name="collectionConfigStruct" type="struct" persistent="false";
```
Comment (jasperboyd) [inline model/entity/AccountCollection.cfc]: Makes sense to me, I just brought these over from Collection.cfc for convenience

File: model/process/AccountCollection_Create.cfc [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	property name="account";
+	property name="collection";
+
+
+	public any function getAccountCollection(){
+		if(!isNull(this.getAccount())){
+			variables.accountCollection.setAccount(this.getAccount());
+		}
+		if(!isNull(this.getCollection())){
+			variables.accountCollection.setCollection(this.getCollection());
```
Comment (rmarchandTen24) [inline model/process/AccountCollection_Create.cfc]: I think that this method should check if the collection is new or not. If getNewFlag then copy over the collectionConfig if persisted then setCollection

File: model/validation/AccountCollection.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+	"properties" : {
+		"collectionDisplayCode":	[{"contexts":"save","uniqueOrNull":true,"regex":"^[a-zA-Z0-9-_.|:~^]+$"}],
+		"collectionConfig":			[{"contexts":"save","required":true}],
```
Comment (rmarchandTen24) [inline model/validation/AccountCollection.json]: collectionConifg is not a required field. It will either be save a collection or a config

Comment (jasperboyd) [inline model/validation/AccountCollection.json]: @rmarchandTen24 the spec says it is required

Comment (sumitverma) [inline model/validation/AccountCollection.json]: it's conditionally required. If there is no collection specified then collectionConfig is required.

Comment (jasperboyd) [inline model/validation/AccountCollection.json]: okay I'll update the spec

Comment (rmarchandTen24) [inline model/validation/AccountCollection.json]: k let me know when you update the validation

File: org/Hibachi/HibachiCollectionService.cfc [hunk 4]
```diff
@@ -377,35 +377,35 @@ component output="false" accessors="true" extends="HibachiService" {
 			baseEntityAlias = '_' & lcase(rc.entityName)
 		};
 
-		if(!isnull(arguments.rc.filterConfig)){
+		if(!isnull(arguments.data.filterConfig)){
```
Comment (rmarchandTen24) [inline org/Hibachi/HibachiCollectionService.cfc]: Change makes sense, but there are still references to rc in the function and the rc is still part of the arguments scope. Just change that to data and make sure all calls to this function don't explicitly say (rc=rc)

Comment (rmarchandTen24) [inline org/Hibachi/HibachiCollectionService.cfc]: ideally make a unit test for this function to make sure we can catch this as a regression in the future

**Pull request merged**